mountpoint-s3-client/tests/head_object.rs

```diff
@@ -105,3 +106,66 @@ async fn test_head_object_no_perm() {
         Err(ObjectClientError::ClientError(S3RequestError::Forbidden(_)))
     ));
 }
+
+// This test relies on s3's expedited object restoration, it takes 1-5 minutes to complete
+#[tokio::test]
+async fn test_head_object_restored() {
+    let sdk_client = get_test_sdk_client().await;
+    let (bucket, prefix) = get_test_bucket_and_prefix("test_head_object_restored");
+
+    let key = format!("{prefix}/hello");
+    let body = b"hello world!";
+    sdk_client
+        .put_object()
+        .bucket(&bucket)
+        .key(&key)
+        .storage_class("GLACIER".into())
+        .body(ByteStream::from(Bytes::from_static(body)))
+        .send()
+        .await
+        .unwrap();
+
+    let client: S3CrtClient = get_test_client();
+    let result = client.head_object(&bucket, &key).await.expect("head_object failed");
+
+    assert_eq!(result.bucket, bucket);
+    assert_eq!(result.object.key, key);
+    assert!(
+        !result.object.restored,
+        "object should become restored only after restoration"
+    );
+
+    sdk_client
+        .restore_object()
+        .bucket(&bucket)
+        .key(&key)
+        .restore_request(
+            aws_sdk_s3::model::RestoreRequest::builder()
+                .set_days(Some(1))
+                .set_glacier_job_parameters(Some(
+                    aws_sdk_s3::model::GlacierJobParameters::builder()
+                        .set_tier(Some(aws_sdk_s3::model::Tier::Expedited))
+                        .build(),
+                ))
+                .build(),
+        )
+        .send()
+        .await
+        .unwrap();
+
+    let timeout_s = 300;
+    let mut i = 0;
+    while i < timeout_s
```

Comment:
I would prefer this
```suggestion
    let timeout_s = 300;
    let start = Instant::now();
    while start.elapsed().as_secs() < timeout_s
```

Reply:
A little more code, since I now need to set `timeout` flag, but more precise in terms of running time. Seems reasonable, fixed.